docs/Blog-Post.md

```diff
@@ -0,0 +1,578 @@
+This is all that is required to shard a model and optimizer across multiple GPUs or TPUs.
+The rest of the training loop remains unchanged.
+You can do fancier things like sharded data loading (which we do in Levanter), but the basic idea is the same.
+
+Tensor parallelism can be added by simply changing the two axis mappings:
+
+```diff
+# Specify which axes we shard for tensor parallelism:
+# specifying "head" shards attention and "mlp" shards the feedforward
++tensor_parallel_mapping = {"head": "model", "mlp": "model"}
+# We store our parameters and optimizer states fully sharded along the embed axis
+-param_mapping = {"embed": "data"}
++param_mapping = {"embed": "data", **tensor_parallel_mapping}
+# During computation, we instead shard our data along the batch axis, and gather the parameters just-in-time
+-data_mapping = {"batch": "data"}
++data_mapping = {"batch": "data", **tensor_parallel_mapping}
+```
+
+## Named Tensors Elsewhere
+JAX already has some built-in support for named tensors in the form of [`xmap`](https://jax.readthedocs.io/en/latest/notebooks/xmap_tutorial.html), which uses something like `vmap`/auto-batching to implement tensors that have both positional and named axes.
+We were initially excited about `xmap` when we first encountered it, but 1) they seem to be deprioritizing it (in favor of `pjit`)
+and 2) ultimately `xmap` can be confusing because you write non-named code for positional axes, then add names "outside"
+of the main model code itself. Ultimately, it's harder reason about this mixed positional/named code (where the axis names are implicit)
+than just using names the whole way through.
+
+Flax supports a logical-to-physical axis mapping thing similar to what's in Haliax. However, the arrays don't carry around
+their axis names, so you have to remember them and pass them in manually when doing partitioning for data parallelism,
+tensor parallism and FSDP. To us, this seems like a bit of a missed opportunity (relative to what we have in Haliax), but it's
+still useful.
+
+Haliax's NamedArrays are probably most similar to [Mesh-Tensorflow](https://github.com/tensorflow/mesh). PyTorch has
+[Named Tensors](https://pytorch.org/docs/stable/named_tensor.html). They're fairly new and "bolted on" to the existing
+positional tensors. They also don't help with model partitioning, which is one of the main use cases for named axes in
+Haliax.
+
+## Haliax Tutorials
+
+This is just a taste of what Haliax can do. For more details, please see our interactive tutorials on Colab:
+
+* [Introduction to Haliax with Transformers](https://colab.research.google.com/drive/1TiTcQQ4V5mopbgCu1SVl-oqJtXn7rFnC?usp=sharing)
+* [Scaling Transformers in Haliax](https://colab.research.google.com/drive/1QX4yH3zRFF3Xiibf1aahETcSQ5nbcUMz?usp=sharing), including FSDP in JAX.
+
+# Levanter: Bitwise Reproducible Foundation Models with JAX
+
+Levanter is a library for training foundation models built on top of Haliax. It provides a complete pipeline
```

Comment:
Add transition: Beyond legibility, the second property is reproducibility...  Let's use the 3 terms (in their noun form) more prominently.